docs/5.1. Data Model - Identifier Mapping.md

```diff
@@ -53,139 +53,136 @@ UUIDs are used throughout the NMOS data model, and can be generated in a number
 
 ### Node ID
 
-For physical Nodes, the Node ID must be universally unique to that Node, and remain the same for all time (much in the same way as a serial number).
+For physical Nodes, the Node ID MUST be universally unique to that Node, and remain the same for all time (much in the same way as a serial number).
 
-For virtual Nodes, the Node ID must be universally unique to that Node. If a snapshot is taken of the virtual Node then it should be stored with the same ID, however the following conditions must be applied when re-deploying the virtual Node from this snapshot:
+For virtual Nodes, the Node ID MUST be universally unique to that Node. If a snapshot is taken of a virtual Node and the snapshot re-deployed:
 
--   If deploying via the same controller which created it and no other instantiation of this virtual Node exists, the original Node ID should be used.
--   If deploying via the same controller which created it and an instantiation of this virtual Node exists, it must be deployed with a new ID. Uses of this Node ID (and child resources) in any other stored data must be mapped to the new ID during this process.
--   If deploying via a different controller it cannot be guaranteed that another instance of this virtual Node isn't already in existence. A new Node ID must therefore be generated, with any stored data being mapped to the new ID during the process.
+- If re-deploying via the same controller which created it and no other instantiation of this virtual Node exists, the original Node ID SHOULD be used.
+- If re-deploying via the same controller which created it and an instantiation of this virtual Node exists, it MUST be deployed with a new ID. Uses of this Node ID (and child resources) in any other stored data MUST be mapped to the new ID during this process.
+- If re-deploying via a different controller and it cannot be guaranteed that another instance of this virtual Node does not already exist, a new Node ID MUST be generated, with any stored data being mapped to the new ID during the process.
 
-In order to simplify the above process for deploying snapshots, you may opt to generate a new Node ID for each instantiation of a virtual Node snapshot, but a mapping service must provide translations between old and new IDs as this may be required to keep session / production / automation data relevant.
+To simplify the above for deploying snapshots, each instantiation of a virtual Node snapshop MAY have a new Node ID. In such a case, a mapping MUST be provided to translate between old and new IDs so that session / production / automation data remain relevant.
```

Comment:
```suggestion
To simplify the above for deploying snapshots, each instantiation of a virtual Node snapshop MAY have a new Node ID. In such cases implementers are encouraged to consider translation between old and new IDs so that session / production / automation data remain relevant.
```

Reply:
Also s/snapshop/snapshot/

Reply:
"each instantiation of a virtual Node snapshot MAY have a new Node ID" taken alone is a little confusing because _concurrent_ instantiations _MUST_ have different Node IDs.

Reply:
Yes. Perhaps we should just simplify lines 58-64 significantly to e.g.:

For virtual Nodes, the Node ID MUST be universally unique to that Node. In cases where it is necessary to redeploy a snapshot of a virtual Node, a new NodeID might be required, together with a mechanism to map between the old and new IDs.

Reply:
Obviously that omits previous "requirements", but it's not clear that they were actually mandated/followed.

Reply:
Makes sense to me

Reply:
Ok, I've done that in 500c75d6a5ed73f4b7aa453b541e727e51d9483a. I'll leave this conversation open for a bit so @andrewbonney sees it.